railties/lib/rails/commands/test.rb

```diff
@@ -1,9 +1,9 @@
 require "rails/test_unit/minitest_plugin"
 
 if defined?(ENGINE_ROOT)
-  $: << File.expand_path("test", ENGINE_ROOT)
+  $LOAD_PATH << File.expand_path("test", ENGINE_ROOT)
 else
-  $: << File.expand_path("../../test", APP_PATH)
+  $LOAD_PATH << File.expand_path("../../test", APP_PATH)
 end
 
-exit Minitest.run(ARGV)
+Minitest.autorun
```

Comment:
this will also fix the double run since autorun guards against being called twice, but it's simpler to do the require :)